specs/updater.go

```diff
@@ -0,0 +1,317 @@
+package specs
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"google.golang.org/api/docs/v1"
+)
+
+// cellBoundaryOffset represents the fixed boundary size between cells
+// in Google Docs API.
+//
+// In the Google Docs API, when calculating cell positions in a table,
+// each cell is separated by a boundary marker. This constant accounts for:
+//   - 1 byte for the cell division character (typically '\n')
+//   - 1 byte for the cell start character (typically '\v')
+//
+// This is used when advancing from one cell's position to the next
+// cell's start index.
+const cellBoundaryOffset int64 = 2
+
+// cellCoordinates represents the position of a status cell in a
+// Google Doc table.
+type cellCoordinates struct {
+	Row int
+	Col int
+}
+
+// updateDocumentStatus updates the Google Doc to change status
+func (r *RejectService) updateDocumentStatus(
+	ctx context.Context,
+	docID string,
+	coords *cellCoordinates,
+	newStatus string,
+) error {
+	doc, err := r.GoogleClient.DocsService.Documents.Get(docID).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("failed to fetch document: %v", err)
+	}
+
+	var table *docs.Table
+	for _, elem := range doc.Body.Content {
+		if elem.Table != nil {
+			table = elem.Table
+			break
+		}
+	}
+	if table == nil {
+		return fmt.Errorf("no metadata table found in document")
+	}
+
+	if coords.Row >= len(table.TableRows) {
+		return fmt.Errorf("row index %d out of bounds (max %d)", coords.Row, len(table.TableRows)-1)
+	}
+	row := table.TableRows[coords.Row]
+
+	if coords.Col >= len(row.TableCells) {
+		return fmt.Errorf("column index %d out of bounds (max %d)", coords.Col, len(row.TableCells)-1)
+	}
+	cell := row.TableCells[coords.Col]
+
+	if len(cell.Content) == 0 || cell.Content[0].Paragraph == nil || len(cell.Content[0].Paragraph.Elements) == 0 {
+		return fmt.Errorf("target cell is empty or malformed")
+	}
+
+	contentElements := cell.Content[0].Paragraph.Elements
+	cellStartIndex := contentElements[0].StartIndex
+	cellEndIndex := contentElements[len(contentElements)-1].EndIndex - 1
+	_, err = r.GoogleClient.DocsService.Documents.BatchUpdate(docID, &docs.BatchUpdateDocumentRequest{
+		Requests: []*docs.Request{
+			{
+				DeleteContentRange: &docs.DeleteContentRangeRequest{
+					Range: &docs.Range{
+						StartIndex: cellStartIndex,
+						EndIndex:   cellEndIndex,
+					},
+				},
+			},
+			{
+				InsertText: &docs.InsertTextRequest{
+					Location: &docs.Location{
+						Index: cellStartIndex,
+					},
+					Text: newStatus,
+				},
+			},
+		},
+	}).Context(ctx).Do()
```

Comment:
At Canonical, we usually use more compact representations for these things. For example, something like:
```go
[]*docs.Request{{
	DeleteContentRange: &docs.DeleteContentRangeRequest{
		Range: &docs.Range{StartIndex: cellStartIndex, EndIndex: cellEndIndex},
	},
}, {
	InsertText: &docs.InsertTextRequest{
		Location: &docs.Location{Index: cellStartIndex},
		Text:     newStatus,
	},
}}
```

This also passes `go fmt`, it's just a more compact representation.